Mapping/utils.py

```diff
@@ -0,0 +1,131 @@
+import ast
+from collections import OrderedDict
+import copy
+import functools
+import textwrap
+from Utility.memo import memoise
+from .types import EV3Type, Local
+
+methods = {}
+setupCode = OrderedDict()
```

Comment:
BTW we could just always include all of the setupcode (imports, device dictionaries, etc) and do away with the stateful part of this. It wouldn't affect performance.

Reply:
There's now _another_ stateful component, global variable allocation. It's currently very hacky, only works in mapping functions, and relies on setupCode